t/t-pointer.sh

```diff
@@ -437,3 +475,224 @@ begin_test "pointer --check (with invalid arguments)"
+  
+  # Test git lfs pointer --file --no-extensions
+  rm -f "$LFSTEST_EXT_LOG"
+  pointer_output="$(git lfs pointer --file=file.dat --no-extensions 2>&1)"
+  
+  # Check that the warning message is NOT present
+  ! echo "$pointer_output" | grep "warning: Using LFS extensions"
+  
+  # Check the pointer does NOT contain extension information
+  ! echo "$pointer_output" | grep "ext-0-caseinverter"
+  
+  # Check the pointer contains standard information only
+  echo "$pointer_output" | grep "oid sha256:$contents_oid"
+  echo "$pointer_output" | grep "size 3"
+  
+  # Verify extension was NOT called (no log file or empty)
+  if [ -f "$LFSTEST_EXT_LOG" ]; then
+    ! grep "clean: file.dat" "$LFSTEST_EXT_LOG"
+  fi
+)
+end_test
+
+begin_test "pointer: extension vs no-extension comparison"
+(
+  set -e
+
+  reponame="pointer-extension-comparison"
+  git init "$reponame"
+  cd "$reponame"
+
+  git lfs track "*.dat"
+  
+  setup_case_inverter_extension
+
+  contents="abc"
+  contents_oid="$(calc_oid "$contents")"
+  inverted_contents="$(invert_case "$contents")"
+  inverted_contents_oid="$(calc_oid "$inverted_contents")"
+  
+  printf "%s" "$contents" >file.dat
+  
+  # Generate pointer with extension
+  rm -f "$LFSTEST_EXT_LOG"
+  git lfs pointer --file=file.dat 2>/dev/null | grep -v "Git LFS pointer" | grep -v "warning:" > pointer-with-ext.txt
+  
+  # Generate pointer without extension
+  rm -f "$LFSTEST_EXT_LOG"
+  git lfs pointer --file=file.dat --no-extensions 2>/dev/null | grep -v "Git LFS pointer" > pointer-no-ext.txt
+  
+  # Verify the pointers are different
+  ! diff pointer-with-ext.txt pointer-no-ext.txt
+  
+  # Verify extension pointer has the ext line
+  grep "ext-0-caseinverter sha256:$contents_oid" pointer-with-ext.txt
+  grep "oid sha256:$inverted_contents_oid" pointer-with-ext.txt
+  
+  # Verify no-extension pointer has standard oid
+  ! grep "ext-0-caseinverter" pointer-no-ext.txt
+  grep "oid sha256:$contents_oid" pointer-no-ext.txt
+)
+end_test
+
+begin_test "pointer: --file --stdin with extension"
+(
+  set -e
+
+  reponame="pointer-extension-file-stdin"
+  git init "$reponame"
+  cd "$reponame"
+
+  git lfs track "*.dat"
+  
+  setup_case_inverter_extension
+
+  contents="abc"
+  contents_oid="$(calc_oid "$contents")"
+  inverted_contents="$(invert_case "$contents")"
+  inverted_contents_oid="$(calc_oid "$inverted_contents")"
+  
+  printf "%s" "$contents" >file.dat
+  
+  # Generate the pointer with extension (this is our expected format)
+  expected_pointer="$(git lfs pointer --file=file.dat 2>/dev/null | grep -v 'Git LFS pointer' | grep -v 'warning:')"
+  
+  # Test --file --stdin comparison (should match)
+  rm -f "$LFSTEST_EXT_LOG"
+  output=$(printf "%s\n" "$expected_pointer" | git lfs pointer --file=file.dat --stdin 2>&1)
+  
+  echo "$output" | grep "Git LFS pointer for file.dat"
+  echo "$output" | grep "ext-0-caseinverter sha256:$contents_oid"
+  echo "$output" | grep "Pointer from STDIN"
```

Comment:
This all works just fine as it stands, so if you'd like to ignore my suggestion, that's perfectly OK!

Aside from dropping the `rm -f "$LFSTEST_EXT_LOG"` line, my only comment is that we can simplify these checks a bit by writing and reading files instead of using shell variables.

For instance, the following seems to work for me, and of course one can then also use `grep "Pointers do not match" pointer.log && exit 1` in the line below.

```suggestion  # Generate the pointer with extension (this is our expected format)
  git lfs pointer --file=file.dat 2>/dev/null | grep -v '\(Git LFS pointer\|warning:\)' >expected-pointer.txt

  # Test --file --stdin comparison (should match)
  cat expected-pointer.txt | git lfs pointer --file=file.dat --stdin 2>&1 | tee pointer.log

  grep "Git LFS pointer for file.dat" pointer.log
  grep "ext-0-caseinverter sha256:$contents_oid" pointer.log
  grep "Pointer from STDIN" pointer.log
```